src/rust/scheduler/handle.rs

```diff
@@ -69,3 +123,16 @@ impl Drop for SchedulerHandle {
         }
     }
 }
+
+impl Hash for SchedulerHandle {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.key.unwrap().hash(state);
```

Comment:
Why is it safe to unwrap here? Could you change unwrap for expect() and add some panic message?